cnap/core/keybroker.py

```diff
@@ -0,0 +1,201 @@
+"""A Keybroker module.
+
+This module provides an object-oriented design for key broker client to connect with
+key broker server (KBS), get model decryption key from the KBS.
+
+Classes:
+    KeyBrokerClientBase: An abstract base class for key broker client.
+    SimpleKeyBrokerClient: A concrete class implementing the KeyBrokerClientBase to connect
+      a simple KBS.
+"""
+
+from abc import ABC, abstractmethod
+import base64
+import logging
+import struct
+import requests
+
+from ccnp import Quote
+from cryptography.hazmat.primitives import hashes
+from cryptography.hazmat.primitives import serialization
+from cryptography.hazmat.primitives.asymmetric import rsa
+from cryptography.hazmat.primitives.asymmetric import padding
+from cryptography.hazmat.primitives.ciphers import Cipher, algorithms, modes
+
+LOG = logging.getLogger(__name__)
+
+# Set the connection timeout to 10s
+TIMEOUT = 10
+
+class KeyBrokerClientBase(ABC):
+    """An abstract base class for key broker client.
+
+    This class serves as a blueprint for subclasses that need to implement
+    `get_key` methods for different types of KBS.
+    """
+
+    @abstractmethod
+    def get_key(self, server_url: str, key_id: str):
+        """Get a key from KBS.
+
+        This method is used to get a key from KBS.
+
+        Args:
+            server_url (str): The url of KBS.
+            key_id (str): The id of the key.
+
+        Raises:
+            ValueError: If the server_url or key_id is None.
+            RuntimeError: If get quote or get key failed.
+            NotImplementedError: If the subclasses don't implement the method.
+        """
+        raise NotImplementedError("Subclasses should implement get_key() method.")
+
+
+class SimpleKeyBrokerClient(KeyBrokerClientBase):
+    """A implementation for key broker client.
+
+    This class implement `get_key` in `KeyBrokerClientBase` abstract base class to connect
+    to a simple KBS.
+    Here is an example flow of a simple KBS:
+        * Accept a quote and a public key from client.
```

Comment:
It's better to use Seq# to replace "*".
Same comment to the follow description below.

Reply:
How about changing it to `-`, it should be generated as a bullet point list in docstring.